reVX/plexos/plexos_plants.py

```diff
@@ -0,0 +1,1006 @@
+                     .format(bus_coords))
+        sc_coords = np.radians(sc_table[['latitude', 'longitude']].values)
+
+        # Filter SC table to points within 'dist_tresh' of coords
+        dist = PlexosPlants._haversine_dist(bus_coords, sc_coords)
+        dist_thresh = \
+            PlexosPlants._substation_distance(sc_table,
+                                              percentile=dist_percentile)
+        logger.debug("- Using distance threshold of {} km".format(dist_thresh))
+        while True:
+            mask = dist <= dist_thresh
+            plant_sc = sc_table[['latitude', 'longitude', lcoe_col]].copy()
+            plant_sc = plant_sc.loc[mask]
+            if len(plant_sc) > 1:
+                break
+            else:
+                dist_thresh *= 1.2
+
+        # Find lowest lcoe site
+        pos = np.argmin(plant_sc[lcoe_col])
+        lcoe_thresh = plant_sc.iloc[pos][lcoe_col] * lcoe_thresh
+        plant_coords = \
+            plant_sc.iloc[pos][['latitude', 'longitude']].values.astype(float)
+        logger.debug("- Plant will be centered at {}".format(plant_coords))
+        logger.debug("- Only supply curve points with an lcoe < {} will be "
+                     "used".format(lcoe_thresh))
+
+        # Filter SC table to lcoe values within 'lcoe_thresh' of min LCOE value
+        sc_cols = ['sc_gid', lcoe_col]
+        plant_sc = sc_table[sc_cols].copy()
+        plant_sc["bus_dist"] = dist
+        mask = plant_sc[lcoe_col] <= lcoe_thresh
+        plant_sc = plant_sc.loc[mask]
+        sc_coords = sc_coords[mask]
+
+        # Sort by distance
+        plant_sc['dist'] = PlexosPlants._haversine_dist(plant_coords,
+                                                        sc_coords)
+        plant_sc = plant_sc.sort_values('dist')
+
+        return plant_sc.reset_index(drop=True)
+
+    @staticmethod
+    def _identify_plants(plant_table, sc_table, dist_percentile=90,
+                         lcoe_col='total_lcoe', lcoe_thresh=1.3,
+                         max_workers=None, plants_per_worker=40):
+        """
+        Identify plant associated with each bus and return supply curve table
+
+        Parameters
+        ----------
+        dist_percentile : int, optional
+            Percentile to use to compute distance threshold using sc_gid to
+            SubStation distance , by default 90
+        lcoe_col : str, optional
+            LCOE column to sort by, by default 'total_lcoe'
+        lcoe_thresh : float, optional
+            LCOE threshold multiplier, exclude sc_gids above threshold,
+            by default 1.3
+        max_workers : int, optional
+            Number of workers to use for plant identification, 1 == serial,
+            > 1 == parallel, None == parallel using all available cpus,
+            by default None
+        plants_per_worker : int, optional
+            Number of plants to identify on each worker, by default 40
+
+        Returns
+        -------
+        plants : list
```

Comment:
I think this could be more descriptive... I think i know what this is but only because we talked through all the nonsense required to associate SC points with plexos plants. I think I'll have no idea what this is in a few months.

Reply:
done